OpenRA.Mods.Common/Traits/Render/WithAimAnimation.cs

```diff
@@ -0,0 +1,75 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	public class WithAimAnimationInfo : ConditionalTraitInfo, Requires<WithSpriteBodyInfo>, Requires<AttackBaseInfo>
+	{
+		[Desc("Armament name")]
+		public readonly string Armament = "primary";
+
+		[Desc("Displayed while targeting.")]
+		[FieldLoader.Require]
+		[SequenceReference] public readonly string Sequence = null;
+
+		[Desc("Which sprite body to modify.")]
+		public readonly string Body = "body";
+
+		public override object Create(ActorInitializer init) { return new WithAimAnimation(init, this); }
+
+		public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			var matches = ai.TraitInfos<WithSpriteBodyInfo>().Count(w => w.Name == Body);
```

Comment:
Style nit: `.Count` requires walking the entire enumeration, which is wasteful in cases where we don't need to know what the actual count is.  When we want to know if there is exactly one it is conceptually better to use `SingleOrDefault` and then check for null.  In practice this will usually walk the entire enumeration too, but it can return early as soon as it finds a second element that matches the condition.